### PR TITLE
Add missing support for StInnerSlotChkUndecl, StInnerObjSlotChkUndecl

### PR DESCRIPTION
**Pull request by pleath**
Add missing support for StInnerSlotChkUndecl, StInnerObjSlotChkUndecl. Obscure opcodes that implement writes to closure-captured block-scoped data requiring dynamic check for ref-before-decl in their home function. I.e., let/const/class declared in switch case.

### CODE REVIEW

**Comment by pleath** [comment]
@rajatd @LouisLaf

**Comment by rajatd** [comment]
LGTM.

File: test/LetConst/tdz2.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
+// Test TDZ behavior when writing to let declared in switch and closure-captured.
```
Comment (akroshg) [inline test/LetConst/tdz2.js]: copyright is missing.

**Comment by LouisLaf** [comment]
LGTM too.

**Comment by pleath** [comment]
Thanks. Added the copyright.

**Pull request merged**